File: app/src/main/java/posidon/launcher/feed/notifications/Notification.java
package posidon.launcher.feed.notifications;

import android.app.PendingIntent;
import android.graphics.drawable.Drawable;

import androidx.annotation.Nullable;

public class Notification {
	public final CharSequence title;
	public final CharSequence text;
	public final boolean isSummary;
	public final Drawable icon;
	public final android.app.Notification.Action[] actions;
	public final CharSequence style;
	public final String key;

	private final PendingIntent contentIntent;

	public Notification(CharSequence title, CharSequence text, boolean isSummary, Drawable icon, android.app.Notification.Action[] actions, PendingIntent contentIntent, @Nullable CharSequence style, String key) {
		this.title = title;
		this.text = text;
		this.isSummary = isSummary;
		this.icon = icon;
		this.actions = actions;
		this.contentIntent = contentIntent;
		this.style = style == null ? null : style.subSequence(25, style.length());
		this.key = key;
	}

	public void open() { try { contentIntent.send(); } catch (Exception ignore) {} }
}

File: app/src/main/java/posidon/launcher/feed/notifications/Notification.kt
package posidon.launcher.feed.notifications

import android.app.Notification
import android.app.PendingIntent
import android.graphics.drawable.Drawable

class Notification(
    val title: CharSequence?,
    val text: CharSequence?,
    val isSummary: Boolean,
    val bigPic: Drawable?,
    val icon: Drawable?,
    val actions: Array<Notification.Action>?,
    private val contentIntent: PendingIntent?,
    val key: String,
    val progress: Float
) {
    fun open() {
        try { contentIntent?.send() }
        catch (e: Exception) {}
    }

    fun cancel() {
        NotificationService.instance.cancelNotification(key)
    }

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (javaClass != other?.javaClass) return false
        other as posidon.launcher.feed.notifications.Notification
        if (title != other.title) return false
        if (isSummary != other.isSummary) return false
        if (key != other.key) return false
        return true
    }

    override fun hashCode(): Int {
        var result = title?.hashCode() ?: 0
        result = 31 * result + isSummary.hashCode()
        result = 31 * result + (actions?.contentHashCode() ?: 0)
        result = 31 * result + (contentIntent?.hashCode() ?: 0)
        result = 31 * result + key.hashCode()
        result = 31 * result + progress.toInt()
        return result
    }
}